### PR TITLE
IT Create Categories Form

### PR DESCRIPTION
**Pull request by mandyfung**
* Tested on IE11
* Integrated with the POST /categories endpoint
* Created new tab called "Manage Categories", it lists all the existing categories first and then an option to create a new category

### CODE REVIEW

**Comment by alekhrycaiko** [comment]
Design:

Would it be possible to make the Add new category text more distinguishable? Perhaps bigger, text, more bold, text centre?

Below are some usability comments, if you can address these to some degree or another that'd be great. They probably aren't simple problems (for the most part), so just try to do what you have time for (whatever seems reasonable):

I'm currently unable to close your modal's with the X button. There's also a flicker present, if you can engineer or design a way to prevent the flicker, that'd be cool (Especially since we can probably reuse that on the history modals).

On success of a submission, the page does a full page reload and I was unable to see a success message except for a brief glimpse.

File: scripts/js/admin_create_category.jsx [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+      return <Checkbox onChange={() => this.toggleCheckbox(label, key)} checked={true} key={key}>{label}</Checkbox>
+    }
+
+    createCheckboxes(e) {
+        let selected;
+        try {
+          selected = e.target.options[e.target.selectedIndex];
+        } catch (err) {
+          selected = e.target;
+        }
```
Comment (alekhrycaiko) [inline scripts/js/admin_create_category.jsx]: what is being thrown here?

File: scripts/js/admin_create_category.jsx [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+                  </Col>
+                </Row>
+                <p/>
+                <ControlLabel><b>Selected Items</b></ControlLabel>
+                  <div style={{margin : "5px", "border-radius": "4px", border: "2px solid #D3D3D3", padding: "10px"}}>
+                    {this.selectedItemsCheckboxes}
+                  </div>
+                  <p/><p/>
+                  <Button type="submit" className="submitBtn" disabled={!this.state.canSubmit}>
+                        Submit Request
```
Comment (alekhrycaiko) [inline scripts/js/admin_create_category.jsx]: When submitted,

Can you please add something like...

{this.state.canSubmit} === true ? Submit Request : Submitting request ...

I'd like all of our buttons to start having this to handle delay while things are being submitted.

File: scripts/js/admin_create_category.jsx (hunk 1 above)
Comment (mandyfung) [inline scripts/js/admin_create_category.jsx]: This was an issue with IE vs Chrome, for some odd reasons on Chrome it couldn't find the options in the target

File: scripts/js/admin_create_category.jsx (hunk 2 above)
Comment (mandyfung) [inline scripts/js/admin_create_category.jsx]: Sounds good, I'll add that in

**Comment by mandyfung** [comment]
> Would it be possible to make the Add new category text more distinguishable? Perhaps bigger, text, more bold, text centre?
Yup, made it bigger and bold, but I think it should remain aligned left to match with with the rest of the page or else I think it looks kinda weird

> I'm currently unable to close your modal's with the X button.
Fixed the issue with closing the modal with the X button

> On success of a submission, the page does a full page reload and I was unable to see a success message except for a brief glimpse.
I tried fixing this, but for some reasons I can't get the success message to stay for a while before reloading and I'm not sure how to add a delay before reloading.

File: scripts/js/admin_create_category.jsx (hunk 1 above)
Comment (alekhrycaiko) [inline scripts/js/admin_create_category.jsx]: Ah, yeah every browser uses different types of event properties so that'd make a lot of sense.

**Pull request closed without merging**